src/main/java/uk/gov/hmcts/reform/pip/data/management/errorhandling/GlobalExceptionHandler.java

```diff
@@ -74,4 +77,19 @@
         return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(exceptionResponse);
     }
 
+    @ExceptionHandler(BlobStorageException.class)
+    public ResponseEntity<ExceptionResponse> handle(BlobStorageException ex) {
```

Comment:
So for this one, the BlobStorageException is built into Azure, not something I wrote. 

I deliberately separated out the 404 because I could reproduce a 404 on demand. I wasn't sure how to create any other type of BlobStorageException on demand, but I didn't want them erroneously characterised as a 404. 

In response to Chris's comment below, I've stopped using the string value comparison and have used a BlobErrorCode comparison instead, which should be more stable.

Does this make sense?